actions/README.md

```diff
@@ -0,0 +1,102 @@
+  ```java
+  provider.get("fetch-user-with-cb").apply(fragmentContext, resultHandler)
+  ```
+ 
+See the [Core module documentation](https://github.com/Knotx/knotx-fragments/tree/master/action/core) 
+for a full list of [actions](#action).
+
+## How does it work?
+[Action](#action) is ready to use [fragment operation](https://github.com/Knotx/knotx-fragments/tree/master/api#fragment-operation) 
+created by an [action factory](#action-factory). Action may be parametrized with JSON (`config`). 
+You can use actions' factories to create actions, however, we recommend using the [action provider](#action-provider) 
+that would hide the complexity of initialization (maintaining stateful actions, combining actions 
+with their behaviours). 
+
+### Action
+Action is a single [fragment operation](https://github.com/Knotx/knotx-fragments/tree/master/api#fragment-operation) 
+that operates on a [Fragment](https://github.com/Knotx/knotx-fragments/tree/master/api#fragment).
+Action can invoke an API and save the response body in a fragment payload or simply modify a fragment's body.
+
+Action may be decorated with [behaviours](https://github.com/Knotx/knotx-fragments/tree/master/action/core#behaviours). 
+Behaviour is a special action that wraps the original action and adds some functionality e.g 
+employ the circuit breaker pattern. 
+
+Please note that actions can be **stateless** and **stateful**.
+
+### Knot
+**Knot** is a scalable **Action** that is available on the [Vert.x Event Bus](https://vertx.io/docs/vertx-core/java/#event_bus). 
+See the [Template Engine](https://github.com/Knotx/knotx-template-engine/blob/master/core/src/main/java/io/knotx/te/core/TemplateEngineKnot.java) 
+module as reference.
+
+### Action provider
+Action provider initializes actions, combines actions with behaviours and caches stateful ones. It
+uses aliases to identify a particular action. Each alias points to action factory name, its configuration.
+If action is a behaviour, it points also to wrapped action (also by alias).
+ 
+Action factories register inside the provider by a [Java Service Loader](https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html).
+ 
+Once configured, the action provider provides actions by alias e.g.
+```hocon
+Action actionInstance = provider.get(ACTION_ALIAS);
+```
+
+Action provider is initialized with an alias to [action factory options](#action-factory-options) map. 
+See the sample configuration below:
+```hocon
+actions {
+  ALIAS_A: ACTION_FACTORY_OPTIONS_A
+  ALIAS_B: ACTION_FACTORY_OPTIONS_B
+}
+```
+
+### Action factory options
+Action factory options contains:
+- [action factory](#action-factory) name
+- action JSON configuration
+- `doAction` for behaviours
+
+and allows:
+- connecting a action factory with configuration
+- linking actions with behaviours.
+
+For more details [here](https://github.com/Knotx/knotx-fragments/blob/master/action/core/docs/asciidoc/dataobjects.adoc#actionfactoryoptions).
```

Comment:
I think it would sound nice like `For more details look [here]`, or maybe just `More details [here]`?